### PR TITLE
config: remove unused secrets for capv and add janitor user

### PR DESCRIPTION
**Pull request by chrischdi**
Follow up to:

* #32619

Related to:
* https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/issues/3040

### CODE REVIEW

**Review by fabriziopandini — COMMENTED** [review]
/lgtm
/approve

**Comment by sbueringer** [comment]
/lgtm

@chrischdi please update the PR title

File: config/prow/cluster/build/build_kubernetes-external-secrets_customresource.yaml [hunk 1]
```diff
@@ -72,9 +72,6 @@ spec:
   backendType: gcpSecretsManager
   projectId: cluster-api-provider-vsphere
   data:
-  - key: capv-gcs-keyfile_json
-    name: capv-gcs-keyfile.json
-    version: latest
   - key: vmc-capv-services_kubeconfig
     name: vmc-capv-services.kubeconfig
     version: latest
```
Comment (sbueringer) [inline config/prow/cluster/build/build_kubernetes-external-secrets_customresource.yaml:77]:
```suggestion
```
@chrischdi let's also drop this one now

**Comment by sbueringer** [comment]
/assign @ameukam 

Context: we are cleaning up our VMC environment so it already works the same way as the new env. This should speed up the migration, more context:
* https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/issues/3040
* https://github.com/kubernetes/k8s.io/issues/6877

**Comment by ameukam** [comment]
/approve
/hold

Remove the hold when ready.

**Comment by sbueringer** [comment]
Thx!

/hold cancel

@ameukam I don't remember. Is this getting deployed automatically, or do we have to ping someone?

**Comment by chrischdi** [comment]
> Thx!
> 
> /hold cancel
> 
> @ameukam I don't remember. Is this getting deployed automatically, or do we have to ping someone?

We will see this in the audit logs of the secrets :-)

**Comment by sbueringer** [comment]
Would be still interesting to know, as it also might tell us who to ping :)

**Comment by chrischdi** [comment]
According the secret logs: the new ones get requested, the old ones stopped getting requested since 10:51 CEST, which matches ~ when this PR merged.

**Pull request merged**